vassal-app/src/main/java/VASSAL/build/module/map/Flare.java

```diff
@@ -446,14 +446,14 @@ public String encode(final Command c) {
    * @return Flare Command object
    */
   public Command decode(final String s) {
-    if (s.startsWith(COMMAND_PREFIX)) {
-      SequenceEncoder.Decoder sd = new SequenceEncoder.Decoder(s, ',');
-
+    if (s.startsWith(COMMAND_PREFIX + getMap().getId())) {
+      SequenceEncoder.Decoder sd = new SequenceEncoder.Decoder(s, '\t');
+      sd.nextToken(); // Skip over the Command Prefix
+      sd.nextToken(); // Skip over the Id
```

Comment:
This is the more common pattern for handling Commands to unique, but similar components. Putting the ID as the first component makes it quicker and easy to check this is the right component. Note the '\t' as the SequenceEncode delimiter, You had use colon in your command prefix and comma for you command components. They all have to match.

Reply:
Also note the need to skip over the Command Prefix and the Id to get to the x,y co-ords.